### ISSUE TITLE
Skip pushing layers when working with docker daemon running in an LXC container

### ISSUE THREAD

**Issue by corneliusweig** [GoogleContainerTools/skaffold#1158]
### Expected behavior
I am using a local kubernetes installation in LXC. This is similar to minikube in that I build docker images directly in my docker daemon. Citing the documentation:

> When using local Kubernetes clusters like Minikube or Docker for Desktop, the push is skipped because the image is already available locally.

When running `skaffold build`, it should know that I am building the containers directly in my local docker daemon and images should not be pushed.

### Actual behavior
Skaffold tries to push the docker images to a remote docker registry.
### Information

- Skaffold version: 0.16.0 and earlier
- Operating system: Linux

### Steps to reproduce the behavior

1. Set up kubectx and docker environment variables to refer to my local kubernetes cluster in an LXC container.
   ```bash
   kubectx lxc
   export DOCKER_TLS_VERIFY=1
   export DOCKER_CERT_PATH=~/.docker-lxc/
   export DOCKER_HOST=tcp://lxc-host:2376
   export DOCKER_API_VERSION=1.37
   ```
2.  Run a skaffold build, for example from the skaffold examples
    ```bash
    cd examples/getting-started
    skaffold build
    ```
This results in
```txt
Sending build context to Docker daemon  3.072kB
Step 1/6 : FROM golang:1.10.1-alpine3.7 as builder
 ---> 52d894fca6d4
Step 2/6 : COPY main.go .
 ---> Using cache
 ---> a2200f79a301
Step 3/6 : RUN go build -o /app main.go
 ---> Using cache
 ---> 8022b9576de0
Step 4/6 : FROM alpine:3.7
 ---> 34ea7509dcad
Step 5/6 : CMD ["./app"]
 ---> Using cache
 ---> 4a267bbcecb0
Step 6/6 : COPY --from=builder /app .
 ---> Using cache
 ---> b84dab284f34
Successfully built b84dab284f34
Successfully tagged d8c3f65f8304b5dac7a26764090809fc:latest
**The push refers to repository [gcr.io/k8s-skaffold/skaffold-example]**
d3e3bbb25860: Preparing
ebf12965380b: Preparing
FATA[0000] build step: building [gcr.io/k8s-skaffold/skaffold-example]: tagging: pushing: unauthorized: You don't have the needed permissions to perform this operation, and you may have invalid credentials. To authenticate your request, follow the steps in: https://cloud.google.com/container-registry/docs/advanced-authentication
```
It is clear that skaffold tries to push the image.

However, the images are already present in my docker daemon:
Both of the follwoing commands works and prints "Hello World!"
```bash
docker run --rm -ti d8c3f65f8304b5dac7a26764090809fc:latest
docker run --rm -ti b84dab284f34
```
and also the layers to be pushed are present, as is evident from
```bash
 docker inspect b84dab284f34 -f '{{.RootFS}}'
{layers [sha256:ebf12965380b39889c99a9c02e82ba465f887b45975b6e389d42e9e6a3857888 sha256:d3e3bbb25860dd91476efc84a11ee94877be5e9e93ebea74262790c1a715ccc3] }
```

**Comment by corneliusweig** [comment]
Alright, I just read the code and found a workaround: if I simply call my kubectl context `docker-for-desktop` when working with my lxc kubernetes, then pushing the image is skipped.

However, I would very much prefer to make this configurable and not bind this behavior to a magic kubectl context name.

**Comment by corneliusweig** [comment]
So even if image pushing becomes configurable, something is still awry. When I do the workaround and name my kubectl context `docker-for-desktop`, then `skaffold dev` does not work as expected.
For example when running example _hot_reload_ with, the pods are created and started but immediately stopped:
```
...
Starting deploy...
kubectl client version: 1.12
pod/node created
pod/python created
Deploy complete in 335.409492ms
Cleaning up...
pod "node" deleted
pod "python" deleted
Cleanup complete in 11.723077741s
FATA[0012] starting logger: initializing pod watcher: unknown (get pods)
```

Looks like the PodWatcher in `pkg/skaffold/kubernetes/log.go` cannot be correctly initialized.

**Comment by nkubala** [comment]
ref https://github.com/GoogleContainerTools/skaffold/issues/1172

**Comment by nkubala** [comment]
yeah there's a lot of weirdness surrounding the "magic" cluster naming. I think the right answer is to have artifact specific push settings in the global config, and then eliminate as much of the `minikube` or `docker-for-desktop` named cluster special casing as we can.

**Comment by corneliusweig** [comment]
Resolved by #1355

**Comment by haf** [comment]
Also, do you have any idea as to why when running minikube with kvm or virtualbox, the docker daemon isn't properly configured to have access? And even if adding a service-account _json_key to the docker daemon in the virtual machine / guest, skaffold is not allowed to do the `pull`, but will reuse images from the guest docker daemon when those have been pulled manually.

I'm commenting on this PR since it would seem that you're working on unifying how the context is managed in Skaffold.
